scene/gui/code_edit.cpp

```diff
@@ -1436,7 +1445,13 @@ bool CodeEdit::is_draw_line_numbers_enabled() const {
 }
 
 void CodeEdit::set_line_numbers_zero_padded(bool p_zero_padded) {
-	p_zero_padded ? line_number_padding = "0" : line_number_padding = " ";
+	String old_line_number_padding = line_number_padding;
+	line_number_padding = p_zero_padded ? "0" : " ";
+	if (line_number_padding == old_line_number_padding) {
+		return;
+	}
```

Comment:
```suggestion
	String new_line_number_padding = p_zero_padded ? "0" : " ";
	if (line_number_padding == new_line_number_padding) {
		return;
	}
	line_number_padding = new_line_number_padding;
```

This can be switched so `line_number_padding` isn't set if it is the same.